apps/test/unit/componentLibrary/CheckboxTest.tsx

```diff
@@ -0,0 +1,120 @@
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import React, {ChangeEvent, useState} from 'react';
+
+import Checkbox from '@cdo/apps/componentLibrary/checkbox';
+
+describe('Design System - Checkbox', () => {
+  const setupCheckbox = (
+    initialChecked: boolean,
+    props: Partial<React.ComponentProps<typeof Checkbox>> = {}
+  ) => {
+    const user = userEvent.setup();
+    const spyOnChange = jest.fn();
+    const Wrapper = () => {
+      const [checked, setChecked] = useState(initialChecked);
+      const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+        setChecked(event.target.checked);
+        spyOnChange(event.target.checked);
+      };
+      return (
+        <Checkbox
+          name="test-checkbox"
+          value="test-value"
+          label="Checkbox label"
+          checked={checked}
+          onChange={handleChange}
+          {...props}
+        />
+      );
+    };
+    render(<Wrapper />);
+    return {user, spyOnChange};
+  };
+
+  const getCheckbox = () => screen.getByRole('checkbox') as HTMLInputElement;
+
+  it('renders with correct label', () => {
+    render(
+      <Checkbox
+        name="test-checkbox"
+        value="test-value"
+        label="Checkbox label"
+        checked={false}
+        onChange={() => null}
+      />
+    );
+    const checkbox = getCheckbox();
+    expect(checkbox).toBeInTheDocument();
+    expect(screen.getByText('Checkbox label')).toBeInTheDocument();
+  });
+
+  it('renders without a label', () => {
+    render(
+      <Checkbox
+        name="test-checkbox"
+        value="test-value"
+        checked={false}
+        onChange={() => null}
+      />
+    );
+    expect(getCheckbox()).toBeInTheDocument();
+    expect(screen.queryByText('Checkbox label')).not.toBeInTheDocument();
+  });
+
+  it('changes checked state on click', async () => {
+    const {user, spyOnChange} = setupCheckbox(false);
+
+    const checkbox = getCheckbox();
+    expect(checkbox.checked).toBe(false);
+
+    await user.click(checkbox);
+    expect(checkbox.checked).toBe(true);
+    expect(spyOnChange).toHaveBeenCalledTimes(1);
+    expect(spyOnChange).toHaveBeenCalledWith(true);
+
+    await user.click(checkbox);
+    expect(checkbox.checked).toBe(false);
+    expect(spyOnChange).toHaveBeenCalledTimes(2);
+    expect(spyOnChange).toHaveBeenCalledWith(false);
+  });
+
+  it('handles indeterminate state correctly', async () => {
```

Comment:
Nit: "handles initial indeterminate state correctly"? I was just doing some Googling, for my learning, what is the use case for the "indeterminate" option?

Reply:
@bencodeorg sorry for long response

> "handles initial indeterminate state correctly"

Not sure why add initial here. Can you explain please?

> what is the use case for the "indeterminate" option?

It's something like 'in bettween'. A good example here is a tree of checkboxes:

[-] Folder 1
  [v] opt a
  [ ] opt b
[ ] Folder 2
  [ ] opt a

[v] - selected checkbox, [-] indeterminate checkbox

Here's more about it on [MDN DOCS](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes)

> 
There are not many use cases for this property. The most common is when a checkbox is available that "owns" a number of sub-options (which are also checkboxes). If all of the sub-options are checked, the owning checkbox is also checked, and if they're all unchecked, the owning checkbox is unchecked. If any one or more of the sub-options have a different state than the others, the owning checkbox is in the indeterminate state.

Reply:
Got it, thanks! Re: test name, never mind, the name is making sense to me at second glance.